rabbitmq/src/main/java/io/micronaut/rabbitmq/intercept/RabbitMQIntroductionAdvice.java

```diff
@@ -372,6 +393,7 @@ private void setBasicProperty(MutableBasicProperties mutableProperties, String n
                         Arrays.stream(method.getArguments())
                                 .filter(arg -> !arg.getAnnotationMetadata().hasStereotype(Bindable.class))
                                 .filter(arg -> !properties.containsKey(arg.getName()))
+                                .filter(arg -> !(arg.getType() == MessageHeaders.class || arg.getType() == RabbitHeaders.class))
```

Comment:
Same as above...

Reply:
You probably are looking at an outdated file. They are removed ithe method in latest code.